### DISCUSSION TITLE
Mark Error On Non-File Nodes

### DISCUSSION THREAD

**Discussion by kapishsingla** [eclipse-theia/theia#14398] [Q&A]
Hi,

I have extended the `FileNavigatorWidget` to serve the purpose of showing the components of document as a non-file Node in Navigator. Now i want to mark the error on node if there is any diagnostic present respect to that node. 

<img width="134" alt="image" src="https://github.com/user-attachments/assets/e0e1ec5c-46f9-43d2-8a32-c065531f45a7">

One of the approach is to get Diagnostic of FileUri under which the non-file Node are created and iterate over the diagnostic and register the uri in `DecoratorService` . This approach seems to be slow. 

Can you help with an another optimized approach?

**Comment by tsmaeder** [comment]
Hi @kapishsingla from looking at the code, I assume you speak of `DecorationsService` you're registering the uri with, right? It's implementation seems to use a cache for the decorations, so there's no a priori reason this should be slow. Could you maybe show a bit more of your code and explain what actions you're performing and what exactly is slow? Is the profiler in the browser developer tools showing any pointers on what is slow?
